### ISSUE TITLE
ruff doesnt work when file doesn't have an extention.

### ISSUE THREAD

**Issue by khufu1** [astral-sh/ruff-vscode#565]
consider this minimal example
`
#!/usr/bin/python3
target = None
if target != None:
    print   ("target's not none")
`
if the file that contains the script has a .py extension it will display the following

![Screenshot from 2024-07-27 01-04-31](https://github.com/user-attachments/assets/88b55cb8-9a84-438d-8690-2f8ed4e64161)

and if i remove the .py extension ,no warnings will display nor file can be formatted even though vscode detects
that it's a python file.

ruff version: 0.5.5
Python version: 3.12.4
vscode version: 1.91.1
OS: archlinux

**Comment by dhruvmanila** [comment]
Yeah, that's happening because a file without an extension is excluded because it's not in the inclusion set. This is a side effect of https://github.com/astral-sh/ruff/commit/0bb2fc6eec875b1fbf06dc25fd9ea9ded4c5cc2b. Maybe in the LSP context, we should look at the [`languageId` in the `TextDocument`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentItem) in case it's not excluded and not present in the inclusion set.
